eng/Analyzers.props

```diff
@@ -2,9 +2,9 @@
   <PropertyGroup>
     <CodeAnalysisRuleset>$(MSBuildThisFileDirectory)CodeAnalysis.ruleset</CodeAnalysisRuleset>
     <!-- Disable analyzers in sourcebuild -->
-    <EnableAnalyzers Condition="'$(DotNetBuildFromSource)' == 'true'">false</EnableAnalyzers>
+    <RunAnalyzers Condition="'$(DotNetBuildFromSource)' == 'true'">false</RunAnalyzers>
```

Comment:
`RunAnalyzers` is the property that is honored by the new analyzers that ship with the SDK, right? Do we need to exclude these from source-build actually? It would be good to now grow the number of differences between a source build and a "normal" build for artificial reasons.

Reply:
> RunAnalyzers is the property that is honored by the new analyzers that ship with the SDK, right?

Yes

> Do we need to exclude these from source-build actually?

I don't know why analyzers were excluded for sourcebuild in the first place.

> It would be good to now grow the number of differences between a source build and a "normal" build for artificial reasons.

This PR isn't growing anything; it's just making it behave the way it was prior to the SDK update with regards to what analyzers run.

If we want to remove the sourcebuild condition, great, I'm all for removing the delta, but it should be done subsequently and by someone who understands why this condition existed in the first place.

Reply:
All tools used by source build have to be part of the source build. I guess the analyzers are excluded to avoid including https://github.com/dotnet/roslyn-analyzers in the source build. (The list of projects currently included in the source build is here: https://github.com/dotnet/source-build/tree/master/repos.)

Reply:
> This PR isn't growing the delta; it's just making it behave the way it was prior to the SDK update with regards to what analyzers run.

Prior to this PR, (new) analyzers that ship with the 5.0 SDK weren't excluded from source-build. You are changing this by combining the condition that we applied on our chosen analyzers with the condition that the SDK uses for the new analyzers. I assume the new ones are source-buildable and if not, then I would assume that this is handled by the SDK.

Reply:
> Prior to this PR, (new) analyzers that ship with the 5.0 SDK weren't excluded from source-build

A week or two ago, before we updated to the new SDK, no analyzers were running here.  When we updated to the new SDK, a few analyzers that are now enabled by default started running (these analyzers existed before, they just weren't enabled before).  I'm returning things to the world as it was when the decision was made to not run any analyzers.

Reply:
> I'm returning things to the world as it was when the decision was made to not run any analyzers.

I don't believe a decision was made to disable all analyzers on source-build. We disabled analyzers that aren't source-buildable: https://github.com/dotnet/corefx/commit/04eba1a41579b3f9d9dbfdc72a96977f5897ff9d. cc @ericstj to share more details.

As said, I would expect analyzers that aren't source-buildable and are running by default, to be excluded from source-build. And that would happen [here](https://github.com/dotnet/roslyn/blob/ff854c695779990b9b269029a8615782a59ec530/src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets#L69) as it wouldn't make sense to require each source-buildable repository to apply that condition.

> A week or two ago, before we updated to the new SDK, no analyzers were running here. When we updated to the new SDK, a few analyzers that are now enabled by default started

That's the behavior change that I was mentioning. Master currently runs these analyzers during source-build and with your PR they won't run anymore. If we would want to enable these again we would need to revert parts of your change, that's why I would like to solve this before merging.

cc @dseefeld

Reply:
> Master currently runs these analyzers during source-build and with your PR they won't run anymore

It hasn't been doing so for years.  It's an unintentional bug that it's been doing so for 5 days.

If you'd like analyzers to run on sourcebuild, changing the policy that's been there for a long time, great.  But it needs to be done thoughtfully and separately.

Reply:
> It hasn't been doing so for years. It's an unintentional bug that it's been doing so for 5 days. This PR fixes that.

You could argue that that these analyzers should then be disabled for non source build builds as well, as they haven't been running before the SDK was updated.

Reply:
> You could argue that that these analyzers should then be disabled for non source build builds as well, as they haven't been running before the SDK was updated.

What analyzers are you referring to?

Reply:
> the decision about which ones to enable did not factor in anything to do with sourcebuild.

Wouldn't that mean that any source-buildable repository will fail to build without this condition as I don't see roslyn-analyzers in the source build list that @jkotas shared? cc @dseefeld

Reply:
> What analyzers are you referring to? The analyzers running by default as part of the SDK are a subset of the analyzers we were previously running in dotnet/runtime for libraries.

That's the part that I wasn't aware of. I thought that analyzers that are now enabled by default are distinct from the ones that we list in Analyzers.props.

Reply:
I don't know, but that's orthogonal to this PR.

Reply:
Ok let's close on this. Thanks for spending time going through this. I think this was a very valuable discussion as it might have identified a source build issue. Let's follow-up with @dseefeld and @ericstj on this.

Reply:
At the time I made that change all those analyzers were prebuilts (shipping versions).  Not sure what the build issue is, but please file an issue if we have one.

Reply:
> That's the part that I wasn't aware of. I thought that analyzers that are now enabled by default are distinct (and source buildable) from the ones that we list in Analyzers.props.

The analyzers that the SDK include are the analyzers included in: `Microsoft.CodeAnalysis.NetAnalyzers`. When this package is included, the SDK is smart enough to use the ones from the package rather than the ones shipped with the SDK. The reason why we went with the package reference instead of using the ones in the SDK, is so that we can consume the analyzers without updating the SDK and have a faster iteration when updating the analyzers in the repo for new rules or bug fixes.